### PR TITLE
crimson: silence warnings from clang++

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test crimson perf

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc [hunk 1]
```diff
@@ -174,7 +174,6 @@ btree_range_pin_t &BtreeLBAManager::get_pin(CachedExtent &e)
       e.cast<LogicalCachedExtent>()->get_pin()).pin;
   } else {
     assert(0 == "impossible");
-    return *static_cast<btree_range_pin_t*>(nullptr);
```
Comment (rzarzynski) [inline src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:177]: How about production builds where the `assert()` above is compiled-out? Maybe `ceph_abort_msg()` instead? It's already marked as `[[noreturn]]`.

File: src/crimson/osd/replicated_recovery_backend.cc [hunk 2]
```diff
@@ -459,7 +459,7 @@ seastar::future<ObjectRecoveryProgress> ReplicatedRecoveryBackend::build_push_op
 				     recovery_info.copy_subset,
 				     progress.data_recovered_to,
 				     available, pop);
-    }).then([this, &recovery_info, &v, &progress, &new_progress, stat, pop]
+    }).then([&recovery_info, &v, &progress, &new_progress, stat, pop]
```
Comment (rzarzynski) [inline src/crimson/osd/replicated_recovery_backend.cc:462]: Sure. This commit LGTM.

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:177]: @rzarzynski thanks, updated and repushed.

**Pull request merged**